components/iosnoop/src/biosnoop.c

```diff
@@ -0,0 +1,62 @@
+#include <uapi/linux/ptrace.h>
+#include <linux/blkdev.h>
+
+struct data_t {
+    u32 pid;
+    u64 rwflag;
+    u64 len;
+};
+
+struct val_t {
+    u32 pid;
+};
+
+BPF_HASH(infobyreq, struct request *, struct val_t);
+BPF_PERF_OUTPUT(events);
+
+int trace_pid_start(struct pt_regs *ctx, struct request *req)
+{
+    struct val_t val = {};
+    u64 ts;
+
+    u64 id = bpf_get_current_pid_tgid();
+    u32 pid = id & (((u64)1 << 32) - 1);
+    u32 tgid = id >> 32;
+    if (tgid != ##TGID##) {
```

Comment:
what's `##TGID##` and what this check does?

Reply:
to filter the io request from tikv process.

Reply:
got it.